python/ray/autoscaler/_private/aws/config.py

```diff
@@ -489,29 +501,70 @@ def _configure_subnet(config):
         ]
         if not subnets:
```

Comment:
```suggestion
        pruned_azs = [s.availability_zone for s in subnets]
        if not subnets:
```
We should capture pruned availability zones as well and throw an error if any were pruned, since we'll just get this less user-friendly error message later otherwise:
```
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/aws/config.py", line 251, in bootstrap_aws
    config = _configure_security_group(config)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/aws/config.py", line 629, in _configure_security_group
    security_groups = _upsert_security_groups(config, node_types_to_configure)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/aws/config.py", line 667, in _upsert_security_groups
    _upsert_security_group_rules(config, security_groups)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/aws/config.py", line 792, in _upsert_security_group_rules
    _update_inbound_rules(sg, sgids, conf)
  File "/Users/pdames/workplace/pdames/ray-project/ray/python/ray/autoscaler/_private/aws/config.py", line 800, in _update_inbound_rules
    target_security_group.authorize_ingress(IpPermissions=ip_permissions)
  File "/Users/pdames/.pyenv/versions/3.7.6/lib/python3.7/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/Users/pdames/.pyenv/versions/3.7.6/lib/python3.7/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(*args, **params)
  File "/Users/pdames/.pyenv/versions/3.7.6/lib/python3.7/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/pdames/.pyenv/versions/3.7.6/lib/python3.7/site-packages/botocore/client.py", line 719, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidGroup.NotFound) when calling the AuthorizeSecurityGroupIngress operation: You have specified two resources that belong to different networks.
```

Reply:
I'm having some trouble consistently reproducing this issue, so don't worry too much about it for now. I'll leave an update here if I can get consistent repro steps and/or a proposed resolution.